packages/env/lib/detect-context.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+/**
+ * External dependencies
+ */
+const util = require( 'util' );
+const fs = require( 'fs' );
+const stream = require( 'stream' );
+const path = require( 'path' );
+
+/**
+ * Promisified dependencies
+ */
+const readDir = util.promisify( fs.readdir );
+const finished = util.promisify( stream.finished );
+
+module.exports = async function detectContext() {
+	const context = {};
+
+	// Race multiple file read streams against each other until
+	// a plugin or theme header is found.
+	const files = ( await readDir( './' ) ).filter( ( file ) => {
+		const extension = path.extname( file );
+		return extension === '.php' || extension === '.css';
```

Comment:
One small optimization for themes would be to look for `style.css` specifically, not just any CSS file.

Reply:
Thanks for taking a look. I changed it here: https://github.com/WordPress/gutenberg/pull/17732/commits/a7f97af35719061ef3656eb7356936b9571fadcd.